app/views/self_registrations/_new_teacher.haml

```diff
@@ -0,0 +1,11 @@
+%h1= t(".title")
+= simple_form_for @resource, :url => self_registrations_path do |form|
+  = form.hidden_field :type
+  = form.input :name
+  = form.input :prename
+  = form.input :email, required: true
+  = form.input :phone
+  = form.association :school, collection: School.all, include_blank: false
```

Comment:
Isn't there some kind of restriction which schools can be chosen by teachers/mentors?

Reply:
Thanks, I've also changed it in mentor form to :mentor kind